### PR TITLE
fix boolean deserialization in config files

### LINKED ISSUE #290: boolean in config file not honored
Boolean configuration parameters in the config file cant be set to `False`, they have to be removed entirely.

### PR DESCRIPTION
**Pull request by HerrMuellerluedenscheid**
resolves #290 

# Pull request checklist

## Bugfix

- [x] A testcase was added to test the behavior

### CODE REVIEW

File: tests/test_configfile.py [hunk 1]
```diff
@@ -211,3 +213,9 @@ def test_config_file_dash_replace(self, input):
             del os.environ['OELINT_CONFIG']
 
             assert getattr(_args, _option.replace('-', '_'))
+
+
+    def test_option_deserialization(self):
+        options = {'a': 'True', 'b': True, 'c': 'False', 'd': False, 'e': 'other value'}
+        deserialized = deserialize_boolean_options(options)
+        assert deserialized == {'a': True, 'b': True, 'c': False, 'd': False, 'e': 'other value'}
```
Comment (priv-kweihmann) [inline tests/test_configfile.py:221]: I see your point in this change.
What I'm not sure about is how this interacts with argparse

how about one of the builtin options? like

```python
parser.add_argument('--nobackup', action='store_true', default=False, help='Don\'t create backup file when auto fixing')
```
Could you please add a testcase that tests that `nobackup=False` in the config file still ends up in nobackup=False in the argparse namespace

**Comment by priv-kweihmann** [comment]
```
E     File "/home/runner/work/oelint-adv/oelint-adv/oelint_adv/__main__.py", line 1
E       from __future__ import annotations
E                                        ^
E   SyntaxError: future feature annotations is not defined
```
doesn't look right

File: tests/test_configfile.py (hunk 1 above)
Comment (priv-kweihmann) [reply tests/test_configfile.py:221]: Sorry my bad - just had a look at the code that is calling this function - it's should work perfectly as proposed

Comment (HerrMuellerluedenscheid) [reply tests/test_configfile.py:221]: I'm not sure I understand. The booleans are all treated the same, no matter if it's applied to `--nobackup` or e.g. `--quiet`. The previous logic was incorrect apparently as `False` provided as a `str` to a `default` with `action=store_true` was `True`.

**Comment by HerrMuellerluedenscheid** [comment]
> ```
> E     File "/home/runner/work/oelint-adv/oelint-adv/oelint_adv/__main__.py", line 1
> E       from __future__ import annotations
> E                                        ^
> E   SyntaxError: future feature annotations is not defined
> ```
> 
> doesn't look right

Right, damn thats only available for python3.7 and later.

File: oelint_adv/__main__.py [hunk 2]
```diff
@@ -33,6 +34,20 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, items)  # pragma: no cover
 
 
+def deserialize_boolean_options(options: dict) -> dict[str, Union[str, bool]]:
```
Comment (priv-kweihmann) [inline oelint_adv/__main__.py:37]: To nitpick I'd say it is `Dict` instead of `dict` in the return type

**Pull request merged**